src/DynamoCoreWpf/Views/FileTrust/FileTrustWarning.xaml.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Threading;
+using Dynamo.Controls;
+using Dynamo.Models;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Controls;
+using Dynamo.Wpf.ViewModels;
+using Dynamo.Wpf.ViewModels.FileTrust;
+
+namespace Dynamo.Wpf.Views.FileTrust
+{
+    /// <summary>
+    /// Interaction logic for RealTimeInfoWindow.xaml
+    /// </summary>
+    public partial class FileTrustWarning : Popup
+    {
+
+        private Window mainWindow;
+        private DynamoViewModel dynViewModel;
+        private FileTrustWarningViewModel fileTrustWarningViewModel;
+
+        public FileTrustWarning(Window dynamoViewWindow)
+        {
+            InitializeComponent();
+
+            mainWindow = dynamoViewWindow;
+            var dynamoView = dynamoViewWindow as DynamoView;
+            if (dynamoView == null) return;
+
+            dynViewModel = dynamoView.DataContext as DynamoViewModel;
+
+            fileTrustWarningViewModel = dynViewModel.FileTrustWViewModel;
+
+            if (fileTrustWarningViewModel == null)
+                fileTrustWarningViewModel = new FileTrustWarningViewModel();
+
+            DataContext = fileTrustWarningViewModel;
+
+            if (dynamoViewWindow == null) return;
+          
+            //Creating the background of the Popup
+            BackgroundRectangle.Rect = new Rect(fileTrustWarningViewModel.PopupBordersOffSet, fileTrustWarningViewModel.PopupBordersOffSet, fileTrustWarningViewModel.PopupRectangleWidth, fileTrustWarningViewModel.PopupRectangleHeight);     
+            
+            fileTrustWarningViewModel.PropertyChanged += ViewModel_PropertyChanged;
+        }
+
+        private void ViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            //When the ShowWarningPopup property is changed we need to enable or disable the Dynamo Run section
+            if (e.PropertyName == "ShowWarningPopup" )
+            {
+                var fileTrustWarningViewModel = sender as FileTrustWarningViewModel;
+                if (fileTrustWarningViewModel == null) return;
+                if (fileTrustWarningViewModel.ShowWarningPopup == true)
+                {
+                    FindPopupPlacementTarget();
+                    //Force to run all the drawing events in the Dispatcher so later we can disable the button/combobox in the Run section
+                    mainWindow.Dispatcher.Invoke(DispatcherPriority.Background, new Action(delegate { }));
+                    DisableRunInteractivity();
+                }                   
+                else
+                    EnableRunInteractivity();
+            }
+        }
+
+        private void FindPopupPlacementTarget()
+        {
+            var bottomBarGrid = mainWindow.FindName("bottomBarGrid") as Grid;
```

Comment:
Just curious, do we have to use `FindName` or we can just use `mainWindow.bottomBarGrid`?

Reply:
@QilongTang updated the type of mainWindow to be DynamoView so we can use all the controls directly as a consequence there is no need to have hard-coded strings.
see the changes in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12916/commits/4505eb7491a99f5ff114cf17fa415da556bef23b